src/Containers/Microsoft.NET.Build.Containers/PlatformMapping.cs

```diff
@@ -0,0 +1,94 @@
+
+            ["linux", "arm64", var _arm64Version ] when isMuslBased => "linux-musl-arm64",
+            ["linux", "arm64", var _arm64Version ] => "linux-arm64",
+
+            ["linux", "arm" ] or ["linux", "arm", "v7" ] when isMuslBased => "linux-musl-arm",
+            ["linux", "arm" ] or ["linux", "arm", "v7" ] => "linux-arm",
+
+            ["linux", "arm", "v6" ] when isMuslBased => "linux-musl-armv6",
+            ["linux", "arm", "v6" ]   => "linux-armv6",
+
+            ["linux", "riscv64" ] when isMuslBased => "linux-musl-riscv64",
+            ["linux", "riscv64" ] => "linux-riscv64",
+
+            ["linux", "ppc64le" ] when isMuslBased => "linux-musl-ppc64le",
+            ["linux", "ppc64le" ] => "linux-ppc64le",
+
+            ["linux", "s390x" ] when isMuslBased => "linux-musl-s390x",
+            ["linux", "s390x" ] => "linux-s390x",
+
+            ["linux", "386" ] when isMuslBased => "linux-musl-x86",
+            ["linux", "386" ] => "linux-x86",
+
+            ["windows", "amd64"] => "win-x64",
+            ["windows", "arm64"] => "win-arm64",
+
+            _ => null // other golang platforms are not supported
+        };
+        return runtimeIdentifier != null;
+    }
+
+    public static bool TryGetDockerPlatformForRid(string runtimeIdentifier, [NotNullWhen(true)] out string? dockerPlatform) {
+        dockerPlatform = null;
+
+        runtimeIdentifier = runtimeIdentifier.Replace("-musl-", "-"); // we lose musl information in the docker platform name
+
+        dockerPlatform = runtimeIdentifier.Split('-') switch {
+            ["linux", "x64"] => "linux/amd64",
+            ["linux", "arm64"] => "linux/arm64",
+            ["linux", "arm"] => "linux/arm/v7",
+            ["linux", "armv6"] => "linux/arm/v6",
+            ["linux", "riscv64"] => "linux/riscv64",
+            ["linux", "ppc64le"] => "linux/ppc64le",
+            ["linux", "s390x"] => "linux/s390x",
+            ["linux", "x86"] => "linux/386",
+            ["win", "x64"] => "windows/amd64",
+            ["win", "arm64"] => "windows/arm64",
+            _ => null
+        };
+        return dockerPlatform != null;
+    }
+
+    public static bool TryGetDockerImageTagForRid(string runtimeIdentifier, [NotNullWhen(true)] out string? dockerPlatformTag) {
+        dockerPlatformTag = null;
+
+        runtimeIdentifier = runtimeIdentifier.Replace("-musl-", "-"); // we lose musl information in the docker platform name
+
+        dockerPlatformTag = runtimeIdentifier.Split('-') switch {
+            ["linux", "x64"] => "amd64",
+            ["linux", "arm64"] => "arm64v8",
+            ["linux", "arm"] => "arm32v7",
+            ["linux", "armv6"] => "arm32v6",
+            ["linux", "riscv64"] => "riscv64",
+            ["linux", "ppc64le"] => "ppc64le",
+            ["linux", "s390x"] => "s390x",
+            ["linux", "x86"] => "386",
+            _ => null // deliberately not trying to make tag names for windows containers
```

Comment:
Can you add support for non-portable rids to this method?
It could be to return `null` for anything that starts with `win`, and return the currently mapped values based on ending with the arch suffix.

An alternative to map from rids could be to map from docker platforms.